recipes/libavif/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+import functools
+import os
+import textwrap
+
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.43.0"
+
+
+class LibAVIFConan(ConanFile):
+    name = "libavif"
+    description = "Library for encoding and decoding .avif files"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/AOMediaCodec/libavif"
+    topics = "avif"
+    license = "BSD-2-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_decoder": ["aom", "dav1d"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_decoder": "dav1d",
+    }
+    generators = "cmake", "cmake_find_package_multi"
+    exports_sources = "CMakeLists.txt", "patches/*"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    @property
+    def _has_dav1d(self):
+        return self.options.with_decoder == "dav1d"
+
+    def requirements(self):
+        self.requires("libaom-av1/3.1.2")
+        self.requires("libyuv/cci.20201106")
+        if self._has_dav1d:
+            self.requires("dav1d/0.9.1")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        root = self._source_subfolder
+        get_args = self.conan_data["sources"][self.version]
+        tools.get(**get_args, destination=root, strip_root=True)
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["AVIF_CODEC_AOM"] = True
```

Comment:
If `self.options.shared` is true, should we pass `-DBUILD_SHARED_LIBS=ON` to the cmake command for libavif?

Reply:
It's handled automatically by Conan.